src/lib.rs

```diff
@@ -507,7 +507,7 @@ fn to_abs_ws_path(path: &Path) -> PathBuf {
             }
         };
 
-        workspace_root.join(path).to_owned()
+        workspace_root.join(path).canonicalize().unwrap()
```

Comment:
Ah, missed this, I think we shouldn't be doing that. `canonicalize` is generally a wrong API to call most of the time, for three reasons: 

* canonicalization is a lossy operation. As a rule of thumb, if the user typed something like `foo/../bar/baz`, it's better to just pass that through as is, and let the underlying OS API to deal with semantics of `..`
* canonicalization is broken on windows -- paths produces by canonicalize on windows tend to not be compatible with many things
* TOCTOU issues -- `canonicalize` is a point-in-time operation, so the results might be different between the time you issue canonicalize request, and the time you actually use the result.

Reply:
Let me send a PR with a fix..